R/utilities-population.R

```diff
@@ -0,0 +1,8 @@
+#' @export
+loadPopulation <- function(fileName, simulation) {

```

Comment:
Note: Loading a population from file requires a simulation. Why? Because we need to be able to check that the file does not contain unknown parameters and also that we can remove the [unit] if part of the exported path, as it is now the default coming from PKSim. For that, we need a simulation to be able to retrieve valid parameter instances

Reply:
I don't really like this. The "population" - which is just a set of parameters - is not really bound to any specific simulation. So in theory I could apply the same population to different simulation.

Also, at the moment, the simulation for which the population is loaded is not stored anywhere, right? So after loading the population for the simulation `sim1` I could apply it (though we still do not know how) to another simulation `sim2`?

> we can remove the [unit] if part of the exported path, as it is now the default coming from PKSim.

I assume that the values coming from PK-Sim are always in default units? In that case, we can just remove the unit part.

Also I do not see a problem if a population contains parameters that are not in the simulation.

Reply:
I agree this is not ideal.  However there is no way otherwise to check the validity of parameters. It becomes important because  your statement `In that case, we can just remove the unit part.` is not accurate

Unfortunately, some parameters have [UNIT] in their name!! (@Yuri05 why is this allowed again) so one cannot simply remove the [xx] part of the name. 

If I remove it all the time, some parameters won't be found. If I never remove it, no parameters will ever be found.

>Also I do not see a problem if a population contains parameters that are not in the simulation.

No that's ok. However a population that only has parameter that do not exist is useless. 

I am not sure what's the best way forward here. In PKSim we don't have the issue because s you alwyas have to select an individual to load a population from csv (which gives us the structure)

@Yuri05 Thoughts?

Reply:
We should not remove any parts from parameter names. If parameter is called "X [unit]" for whatever reason - we have to live with this. In MoBi user can name parameters whatever he likes (unless we e.g. prohibit usage of [])

Reply:
@Yuri05 But PK-Sim exports the population parameters with a `[unit]` suffix:

`"Organism|Weight [kg]"`

So in order to find this parameter in the simulation and change it's value the suffix must be removed...

Reply:
Ah ok, I misunderstood what you meant.
Are **all** parameters exported like that? (e.g. how are dimensionless parameters exported: like `"P []"` or just `"P"`?)
* If yes: just remove **last occurance** of [...] in the parameter name
* If no:
  * first remove **last occurance** of [...] in the parameter name and check if parameter exists in simulation
  * if no: check if parameter with original name (nothing removed) exists

Reply:
The [] must be removed. However the only way to remove it  for sure is to have a simulation has a parameter
But having a simulation to create a population is weird.


>Are all parameters exported like that?

No dimensionsless parameter have no unit exporter

> first remove last occurance of [...] in the parameter name and check if parameter exists in simulation

That's the whole point. Here we need a simulation and it sucks from an API point of view

Reply:
Well, I would _import_ a population without any change of parameter paths.
But **at the time point where a population is applied to a simulation**: I would check and try with and without units.

Reply:
Ok that was my next proposal. In that case, we'll have to split core and pksim implementation however because the behavior in PKSIm will be different.

Reply:
> But at the time point where a population is applied to a simulation: I would check and try with and without units.

Actually I also wanted to propose this..

Reply:
Well we are all in agreement. I'll implement it that way